File: cpp/maxperimeter.cpp
#include <algorithm>
#include <vector>
using namespace std;
class Solution {
public:
    long long largestPerimeter(vector<int>& nums) {
        sort(nums.rbegin(), nums.rend()); 
        

        int start = 0;
        while (start < nums.size()) {
            long sum = 0;
            for (int i = start + 1; i < nums.size(); i++) {
                sum += nums[i]; 
            }
            if (sum <= nums[start]) {
                start += 1;
                continue;
            }
            else{
                return sum + nums[start];
            }
        }
        return -1;
    }
};

File: py/maxperimeter.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        
        start = 0
        while start < len(nums):
            total_sum = 0
            for i in range(start + 1, len(nums)):
                total_sum += nums[i]
            
            if total_sum <= nums[start]:
                start += 1
                continue
            else:
                return total_sum + nums[start]
        
        return -1